src/hooks/vaultsHooks.ts

```diff
@@ -0,0 +1,426 @@
+    queryKey: ['vaultPositions'],
+    queryFn: async () => {
+      return wrapNullable(
+        VaultCalculator.getVaultPositionsResponse(await placeholderFetchMegavaultPositions())
+      );
+    },
+    ...vaultQueryOptions,
+  });
+
+  const calculatedPositions = useMemo(() => {
+    if (vaultPositions?.data == null || marketsMap == null || marketsMap.size === 0) {
+      return undefined;
+    }
+    return VaultCalculator.calculateVaultPositions(
+      vaultPositions.data,
+      subvaultHistories?.data,
+      marketsMap
+    );
+  }, [subvaultHistories, vaultPositions, marketsMap]);
+
+  return calculatedPositions;
+};
+
+const vaultAccountQueryKey = ['vaultAccount'];
+
+export function forceRefreshVaultAccount() {
+  appQueryClient.invalidateQueries({ queryKey: vaultAccountQueryKey, exact: true });
+}
+
+export const useLoadedVaultAccount = () => {
+  const { data: accountVault } = useQuery({
+    queryKey: vaultAccountQueryKey,
+    queryFn: async () => {
+      const [acc, transfers] = await Promise.all([
+        placeholderFetchVaultAccount(),
+        placeholderFetchVaultAccountTransfers(),
+      ]);
+
+      const parsedAccount = VaultAccountCalculator.getAccountVaultResponse(acc);
+      const parsedTransfers = VaultAccountCalculator.getTransfersBetweenResponse(transfers);
+
+      if (parsedAccount == null || parsedTransfers == null) {
+        return wrapNullable(undefined);
+      }
+      return wrapNullable(
+        VaultAccountCalculator.calculateUserVaultInfo(parsedAccount, parsedTransfers)
+      );
+    },
+    ...vaultQueryOptions,
+  });
+  return accountVault?.data;
+};
+
+export const useLoadedVaultAccountTransfers = () => {
+  const account = useLoadedVaultAccount();
+  return useMemo(() => account?.vaultTransfers?.toArray(), [account?.vaultTransfers]);
+};
+
+const useVaultFormAmountDebounced = () => {
+  const amount = useAppSelector((state) => state.vaults.vaultForm.amount);
+  return useDebounceHf(amount, 500);
+};
+
+export const useVaultFormSlippage = () => {
+  const amount = useVaultFormAmountDebounced();
+  const operation = useAppSelector((state) => state.vaults.vaultForm.operation);
+  const vaultBalance = useLoadedVaultAccount();
+
+  const { data: slippageResp } = useQuery({
+    queryKey: [
+      'vaultSlippage',
+      amount,
+      operation,
+      vaultBalance?.balanceUsdc,
+      vaultBalance?.balanceShares,
+    ],
+    queryFn: async () => {
+      if (operation === 'DEPOSIT' || amount.trim() === '') {
+        return wrapNullable(undefined);
+      }
+      const sharesToWithdraw = MustBigNumber(amount).div(
+        // usdc / share to determine share value
+        MustBigNumber(vaultBalance?.balanceUsdc).div(MustBigNumber(vaultBalance?.balanceShares))
+      );
+      const slippage = await placeholderFetchVaultFormSlippage(
+        sharesToWithdraw.decimalPlaces(0, BigNumber.ROUND_FLOOR).toNumber(),
+        MustBigNumber(amount).times(0.96145).toNumber()
```

Comment:
is the `0.96145` going to be dynamic in the future? If not can we pull this out into a more descriptive constant?

Reply:
nope, temporary fake data thing, will remove when we have real calls today hopefully